### PR TITLE
CSUB-1071: Improvements revealed by newer rustc

### PR DESCRIPTION
**Pull request by atodorov**
# Description of proposed changes

a few improvements revealed to us when using a newer rustc while waiting for the issues in #159 to be resolved.

---

Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -484,7 +484,7 @@ impl pallet_babe::Config for Runtime {
     type EpochDuration = EpochDuration;
     type ExpectedBlockTime = ExpectedBlockTime;
     type EpochChangeTrigger = pallet_babe::ExternalTrigger;
-    type KeyOwnerProof = <Historical as KeyOwnerProofSystem<(KeyTypeId, BabeId)>>::Proof;
+    type KeyOwnerProof = sp_session::MembershipProof;
```
Comment (atodorov) [inline runtime/src/lib.rs:487]: See commit log message for more references and links to upstream.

IDK what's the impact of this change though.

**Review by DylanVerstraete — APPROVED** [review]
LGTM

**Review by beqaabu — APPROVED** [review]
`pallet-session` internally uses `sp_session::MembershipProof` either way, so shouldn't have any issues + upstream and everyone else is doing the same.. so it shouldn't be a problem

**Comment by atodorov** [comment]
Sigh, needed to bump the version. Will merge after CI jobs pass. :-(

**Review by beqaabu — APPROVED** [review]

**Review by zacharyfrederick — APPROVED** [review]
rustc knows all

**Pull request merged**